blog/_posts/2021-07-31-Mitchell-Johnson_Jamie-Schmidt.md

```diff
@@ -0,0 +1,118 @@
+
+This means that it is often beneficial to pass Genomes as Heads if you want to be as generic as possible while still allowing full functionality of the Genome.
+
+## DynamicOrg / author: mitchell and jamie, predominantly michell though
+-what is an organism in MABE2 and how it was originally used to create an expiriment
+-how DynamicOrg makes the entire process of designing, implementing, and running an expiriment much easier
+
+In MABE2, an Organism can be thought of as a box which contains some number of Genomes and some number of Brains.  An Organism has three main functions: 
+1. An Organism needs to be able to copy itself to form one or more children.
+2. An Organism needs to be able to mutate itself in some way for evolution to happen.
+3. An Organism needs to return some kind of object which consists of all the data the Evaluator needs. 
+
+Of course, different Organisms can handle things differently, so for example an Organism in MABE (call it a BitOrg) could consist entirely of one TypedGenome<bool>, which is a Genome that is composed entirely of bits.  It can copy that whenever it reproduces, it can flip some random number of bits every time it mutates, and it can simply return the bits of that Genome in a BitVector (a specialized vector of bits which Empirical uses in place of std::BitMaps and std::vector<bool>s).
+  
+But what's a DynamicOrg?  Well, a DynamicOrg is an Organism that is nearly totally built by the config file.  Unlike our previous example, where the BitOrg knew it had a Genome of bits and how to mutate and return it, a DynamicOrg can't.  Ideally, you'd be able to specify what Genomes and Brains were in this DynamicOrg, and that would be it.  Specifying Genomes is relatively easy, since they're all completely self-contained, but the Brains pose a more significant challenge, which is why we'll be discussing them in the section after this.
+
+Now, a DynamicOrg must necessarily be significantly less restrictive.  Copying itself is fine, we just copy all its Genomes and Brains.  Mutations are also reasonably simple, since all we need to do is mutate every Genome in it.  But how on earth do we know what to return?  Simply put, we don't, so we just want to return an emp::DataMap.  The Evaluator is what knows, so we let the Evalutor decide.  The Brains then all edit this emp::DataMap in order, in whatever way they are designed to do.  
+  
+In the config file, this means all a DynamicOrg needs is a list of names of all the Genomes and Brains it will contain.  So long as all these Genomes and Brains are properly named in the confic file, this will work seamlessly. 
+  
+## Brains / author: jamie
```

Comment:
If you want to have authorship attributed for different parts, a kind of new thing that people are doing are “Author Contribution Statements” like this https://www.epj.org/images/stories/faq/examples-of-author-contributions.pdf You could just slot that in at the end